src/burnchains/burnchain.rs

```diff
@@ -157,14 +160,55 @@ impl BurnchainStateTransition {
         let consumed_leader_keys =
             sort_tx.get_consumed_leader_keys(&parent_snapshot, &block_commits)?;
 
+        // assemble the commit windows
+        let mut windowed_block_commits = vec![block_commits];
+        let mut windowed_user_burns = vec![user_burns];
+
+        for blocks_back in 0..(MINING_COMMITMENT_WINDOW - 1) {
+            if parent_snapshot.block_height < (blocks_back as u64) {
+                debug!("Mining commitment window shortened because block height is less than window size";
+                       "block_height" => %parent_snapshot.block_height,
+                       "window_size" => %MINING_COMMITMENT_WINDOW);
+                break;
+            }
+            let block_height = parent_snapshot.block_height - (blocks_back as u64);
+            let sortition_id = match sort_tx.get_block_snapshot_by_height(block_height)? {
+                Some(sn) => sn.sortition_id,
+                None => break,
+            };
+            windowed_block_commits.push(SortitionDB::get_block_commits_by_block(
+                sort_tx.tx(),
+                &sortition_id,
+            )?);
+            windowed_user_burns.push(SortitionDB::get_user_burns_by_block(
+                sort_tx.tx(),
+                &sortition_id,
+            )?);
+        }
+
+        // reverse vecs so that windows are in ascending block height order
+        windowed_block_commits.reverse();
+        windowed_user_burns.reverse();
+
+        // figure out if the PoX sunset finished during the window
+        let window_end_height = parent_snapshot.block_height + 1;
```

Comment:
Does the window get disabled after PoX sunset?

Reply:
No -- it just changes which output it tries to match in the BlockCommitOp for UTXO chaining (because after the sunset completes, the BlockCommitOp has 1 fewer output).